documentation/docs/getting-started/core-concepts.md

```diff
@@ -0,0 +1,153 @@
+
+### Jest
+
+Jest is a testing framework with support for snapshot testing. Fusion.js provides integrated support for Jest.
+
+### Lint
+
+A linter is a tool that statically analyzes code and reports certain patterns of code if they are error-prone or if they deviate from a code style standard.
+
+### Prettier
+
+An opinionated code formatter that ensures code consistency. Similar to Go's gofmt.
+
+### Redux
+
+Redux is a popular state management library that is commonly used with React.
+
+### REST
+
+REST stands for Representational State Transfer. In the context of web development, it refers specifically to HTTP endpoint architectures that leverage HTTP method names and object-oriented endpoint naming (in contrast to RPC, which revolves around verb-oriented endpoint naming).
+
+### RPC
+
+RPC stands for Remote Procedure Call. In the context of web development, it refers specifically to HTTP endpoint architectures revolving around named method definitions (in contrast to REST architectures, which revolve around leveraging HTTP method names and noun-oriented endpoint naming).
+
+### SPA
+
+SPA stands for Single Page Application. In the context of React, it means an application that uses react-router and the HTML5 history.pushState API under the hood for page transitions, as opposed to traditional JSP-style applications where page transitions are accomplished via full page refreshes.
+
+### SSR
+
+SSR stands for Server-Side Rendering. It means rendering a React tree on the server and serving the output as HTML (while also using the same Javascript code in the browser in a single-page application architecture). This can improve time-to-first-paint because browsers' HTML rendering engine is faster than Javascript/DOM-based rendering. Server-side rendering is enabled by default in Fusion.js.
+
+### Token
+
+A token is an identifier for a dependency in Fusion.js' DI system. Typically, you declare that a plugin depends on a token and bind it to a local variable name (e.g. the declaration `{deps: {logger: LoggerToken}}` means to make available a local variable called `logger` whose value is whatever service we specified `LoggerToken` to point to). `LoggerToken` could in turn point to a hypothetical WinstonService or a ConsoleService depending on how you want to configure your application.
+
+### Tree-shaking
+
+Tree-shaking is a compiler optimization that removes unused imports. In Fusion.js, tree-shaking is used to remove server-side code from universal code when generating browser bundles. Note that tree-shaking may not occur if an unused dependency has side effects upon being imported.
+
+### Universal code
+
+Universal code refers to code that lives in a single file but runs on both the server and the browser. Note that it's possible to conditionally run code only on the server or only on the browser via `if (__NODE__) {}` and `if (__BROWSER__) {}` statements, respectively.
+
```

Comment:
Let's link to the universal code doc: https://fusionjs.com/api/fusion-docs/universal-code